doc/modules/feature_extraction.rst

```diff
@@ -101,6 +101,105 @@ memory the ``DictVectorizer`` class uses a ``scipy.sparse`` matrix by
 default instead of a ``numpy.ndarray``.
 
 
+.. _column_transformer:
```

Comment:
This should be in compose.rst, but perhaps noted at the top of this file

Reply:
Yes, I know, but also (related to what I mentioned here: https://github.com/scikit-learn/scikit-learn/pull/9012#issuecomment-352207204):

- when moving to compose.rst, I think we should use a different example (eg using transformers from preprocessing module, as I think that is a more typical use case)
- we should reference this in preprocessing.rst
- we should add a better 'typical data science usecase" example for the example gallery
- I would maybe keep the explanation currently in `feature_extraction.rst` (the example), but shorten it by referring to compose.rst for the general explanation.

I can work on the above this week. But in light of getting this merged sooner rather than later, I would prefer doing it as a follow-up PR, if that is fine? (I can also do a minimal here and simply move the current docs addition to compose.rst without any of the other mentioned improvements).